prime-router/docs/design/proposals/0024-original-message-passthrough.md

```diff
@@ -0,0 +1,172 @@
+      `Report.Format` (e.g. `HL7_ORIGINAL`) or having the `translate` function retrieve sender settings for the original
+      submission.
+
+    * *Pro*: This approach potentially has the least amount of modifications to existing processes and makes use of
+      information that is already stored in the database. There is no need to store additional information about the
+      submission to facilitate the sendOriginal feature.
+    * *Pro*: There is no risk of leaking private information about ReportStream's storage as the FHIR bundle is not
+      modified.
+    * *Con*: This approach is highly reliant on the processes performed at the `receive` pipeline step remaining as they
+      are now. Any refactoring of this code must take sendOriginal into consideration.
+
+## Limitations
+
+* Filtering can still occur as a routing process, but the nature of sending an unmodified HL7v2 file will mean any
+  conditions that would have been eliminated by filtering would be sent regardless.
+* Batching cannot be performed for sendOriginal files.
+* Filenames in the Azure blobstore currently reflect pipeline specific information to more easily identify when and why
+  a file was created.
+    * There is currently a limitation that files must have unique names, and blobURLs cannot be reused.
+    * External filenames are in many cases based on the blobURL presently.
+
+## Feature Design
+
+After performing analysis of the possible implementations above, the **Add feature to the UP translate step: retrieve
+blobURL from report_file DB table** implementation was selected.
+
+The overall design of the sendOriginal feature is as follows:
+
+* A new `Topic` feature flag, `sendOriginal`, will be used for sendOriginal items. The use of Topic to define
+  sendOriginal guarantees that data that must be handled via sendOriginal is not processed as other
+  full ELR data and is only sent to receivers in its original state. This also allows receivers to be able to configure
+  endpoints specific to the data for this particular sender, separate from any batching settings the receiver may
+  otherwise have configured.
+* As the `Topic` is unique, no other identifiers for sendOriginal items are needed. Configuring the sender and
+  receivers for the Topic with `sendOriginal` defined is all that is required.
+* The `translate` step will copy the original blob data from the blobstore instead of translating from the internal
+  FHIR bundle. Enhancements to `BlobAccess` that have been made in preparation for the `Manage Translation Schemas`
+  feature can be utilized to perform this operation. The `translate` step will then directly insert the resulting blob
+  to the `send` queue.
+* `Report.formExternalFilename` uses the filename of the blobstore item, so there should not be any further
+  modifications necessary as long as the blobstore item is named correctly by the `translate` function.
+
+### Issues to be created
```

Comment:
I think the following tickets may be missing, based on the descriptions of the three tickets below:

- Update receive or convert step to verify submitted report only contains a single message when a report is received under a topic with that setting enabled.  We can discuss this during implementation, but imo this should be implemented using something like polymorphism so we don't have "if topic.setting == true" sprinkled everywhere in the future. Same idea goes for translate step.

Reply:
Also, this should probably just be done with the translate step update ticket. And then the first two bullets you have should be done together. While the work can be split up the way you have it, its less overhead/management to combine them.

Reply:
I've added the additional requirement and consolidated the tickets; I will work on updating the tickets as written to reflect this.